client/app/telecom/pack/xdsl/access/comfortExchange/pack-xdsl-access-comfort-exchange.controller.js

```diff
@@ -0,0 +1,81 @@
+import { COMFORT_EXCHANGE_TYPE_ERROR } from './pack-xdsl-access-comfort-exchange.constant';
+
+angular.module('managerApp').controller('XdslAccessComfortExchangeCtrl', class XdslAccessComfortExchangeCtrl {
+  /* @ngInject */
+  constructor($stateParams, $translate, OvhApiXdsl, TucToast, TucToastError) {
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.OvhApiXdsl = OvhApiXdsl;
+    this.TucToast = TucToast;
+    this.TucToastError = TucToastError;
+  }
+
+  /*= =====================================
+  =            INITIALIZATION            =
+  ====================================== */
+  $onInit() {
+    this.exchange = {
+      order: {
+        url: null,
+        id: null,
+        priceWithoutTax: null,
+        priceWithTax: null,
+        tax: null,
+      },
+      isSuccess: false,
+    };
+    this.isAvailable = false;
+    this.getListOpenedRMA();
+  }
+
+  getListOpenedRMA() {
+    this.isRMAOpened = false;
+    return this.OvhApiXdsl.RMA().v6().query({
+      xdslId: this.xdslId,
+    }).$promise.then((result) => {
+      if (result.length > 0) {
+        this.rmas = [];
+        this.isRMAOpened = true;
+        result.forEach(id => this.OvhApiXdsl.RMA().v6().getRMA({
```

Comment:
```suggestion
         return this.$q.all(_.map(rmas, id => this.OvhApiXdsl.RMA().v6().getRMA({
          xdslId: this.xdslId,
        }, { id })))
        .then(rmas => this.rmas = _.map(rmas, rma => {
            creationDateTime: rma.creationDatetime,
            equipmentReference: rma.equipmentReference,
            id: rma.id,
            newMerchandise: rma.newMerchandise,
            status: rma.status,
          })
```

Reply:
@FredericEspiau Are you sure sure about this code ? I think it missed something. Or I don't understand what you are doing here.

Reply:
Sorry but it doesn't as well as expected, so I don't apply it.